### PR TITLE
Report nose test failures to the nose test result.

### LINKED ISSUE #25: Exit status is set to 0 on failure instead of 1
When running tests with nose and the flaky plugin, the exit status was not being set to 1 when the only tests that failed were ones that run with the flaky decorator and had max tries greater than 1.  If a test that was not decorated with `@flaky` also failed, then the exit status was properly set to 1.  

For example, here is a test that always fails, run with nose and without flaky.  The exit status is `1`.
```
$ nosetests test_flaky.py:TestFlaky
F
-----------------------------------------------------------------------------
1) FAIL: test_flaky_exit (test_flaky.TestFlaky)

   Traceback (most recent call last):
    test_flaky.py line 7 in test_flaky_exit
      self.assertTrue(False)
   AssertionError: False is not true


-----------------------------------------------------------------------------
1 test run in 0.0 seconds. 
1 FAILED (0 tests passed)
$ echo $?
1
```

... And here is the same test, run with nose and using flaky with the default 2 max tries.  It correctly runs the test twice and prints the correct results, but then exits with `0` status.
```
$ nosetests test_flaky.py:TestFlaky --with-flaky
-F
-----------------------------------------------------------------------------
1) FAIL: test_flaky_exit (test_flaky.TestFlaky)

   Traceback (most recent call last):
    test_flaky.py line 7 in test_flaky_exit
      self.assertTrue(False)
   AssertionError: False is not true


-----------------------------------------------------------------------------
2 tests run in 0.0 seconds. 
1 FAILED, 1 skipped (0 tests passed)
$ echo $?
0
```

The root cause is that the flaky plugin passes a newly created `TextTestResult` object to `_rerun_test` rather than the original object that nose created, leaving nose unaware of any rerun results.

### PR DESCRIPTION
**Pull request by Jeff-Meadows**
Fixes #25.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @Jeff-Meadows has signed the CLA. Thanks for the pull request!

**Comment by Jeff-Meadows** [comment]
I noticed while testing this PR that flaky had another bug, which I've also fixed in this PR. The bug allowed flaky to retry too many times under certain circumstances, eg @flaky(3, 2) and [fail, fail, pass, pass] would be a pass. This shouldn't happen because that's 4 runs, which is greater than the specified max_runs of 3.

**Comment by clytwynec** [comment]
Hi @Jeff-Meadows,  I just tried this change out and it does fix issue #25,  but it doesn't seem to be retrying the test anymore. See https://gist.github.com/clytwynec/7fffb0f58ac071e6adf8.

**Comment by clytwynec** [comment]
Ok, I just looked at this again.  It looks like it's working correctly, the reporting just looks different than I was used to.

File: flaky/flaky_nose_plugin.py [hunk 1]
```diff
@@ -66,7 +67,10 @@ def handleError(self, test, err):
             `bool`
         """
         # pylint:disable=invalid-name
-        return self._handle_test_error_or_failure(test, err)
+        want_error = self._handle_test_error_or_failure(test, err)
+        if not want_error:
+            self._nose_result.addError(test, err)
```
Comment (jmoldow) [inline flaky/flaky_nose_plugin.py]: This is a new behavior. Can you document it in the docstring?

File: flaky/flaky_nose_plugin.py [hunk 2]
```diff
@@ -85,7 +89,10 @@ def handleFailure(self, test, err):
             `bool`
         """
         # pylint:disable=invalid-name
-        return self._handle_test_error_or_failure(test, err)
+        want_failure = self._handle_test_error_or_failure(test, err)
+        if not want_failure:
+            self._nose_result.addFailure(test, err)
```
Comment (jmoldow) [inline flaky/flaky_nose_plugin.py]: This is a new behavior. Can you document it in the docstring?

File: flaky/flaky_nose_plugin.py [hunk 3]
```diff
@@ -120,6 +127,18 @@ def report(self, stream):
         if self._flaky_report:
             self._add_flaky_report(stream)
 
+    def prepareTestResult(self, result):
+        """
+        Baseclass override. Called right before the first test is run.
```
Comment (jmoldow) [inline flaky/flaky_nose_plugin.py]: This doesn't describe what the handler is actually going to end up doing.

**Pull request merged**